src/analyses/apron/relationAnalysis.apron.ml

```diff
@@ -202,8 +209,8 @@ struct
       (* TODO: don't go through CIL exp? *)
       let e1 = BinOp (Le, Lval (Cil.var x), (Cil.kintegerCilint ik type_max), intType) in
       let e2 = BinOp (Ge, Lval (Cil.var x), (Cil.kintegerCilint ik type_min), intType) in
-      let rel = RD.assert_inv rel e1 false (no_overflow ask e1) in (* TODO: how can be overflow when asserting type bounds? *)
-      let rel = RD.assert_inv rel e2 false (no_overflow ask e2) in
+      let rel = RD.assert_inv ask rel e1 false (no_overflow ask e1) in (* TODO: how can be overflow when asserting type bounds? *)
```

Comment:
We also thought about that and tried to implement it by calling `no_overflow ...` in `texpr1_of_cil_exp`, where we use `ask` also for the analysis of the casts in the expression. This results in unsound behaviour, because in the file `relationalAnalysis` the function `no_overflow` is very often called with a different expression than the one that is passed as a parameter to the `Convert` module. 
In order to analyze the overflow behaviour we always need the original expression, but in many functions the original expression is simplified before being passed as a parameter to `texpr1_of_cil_exp`. So we need the `no_ov` flag anyway.